browser/resources/settings/brave_content_page/containers.ts

```diff
@@ -0,0 +1,136 @@
+
+const SettingsBraveContentContainersElementBase = I18nMixin(PolymerElement)
+
+/**
+ * 'settings-brave-content-containers' is the settings page containing settings for Containers
+ */
+export class SettingsBraveContentContainersElement extends SettingsBraveContentContainersElementBase {
+  private browserProxy: ContainersSettingsPageBrowserProxy =
+    ContainersSettingsPageBrowserProxy.getInstance()
+
+  static get is() {
+    return 'settings-brave-content-containers'
+  }
+
+  static get template() {
+    return getTemplate()
+  }
+
+  static get properties() {
+    return {
+      containersList_: {
+        type: Array as unknown as Container[],
+      },
+      editingContainer_: Object as unknown as Container | null,
+      deletingContainer_: Object as unknown as Container | null,
+      editDialogTitle_: {
+        type: String,
+        computed: 'computeEditingDialogTitle_(editingContainer_)',
+      },
+    }
+  }
+
+  declare containersList_: Container[]
+  declare editingContainer_: Container | null
+  declare deletingContainer_: Container | null
+  declare editDialogTitle_: string
+
+  override ready() {
+    super.ready()
+    this.browserProxy.handler.getContainers().then(({ containers }) => {
+      this.onContainersListUpdated_(containers)
+    })
+    this.browserProxy.callbackRouter.onContainersChanged.addListener(
+      this.onContainersListUpdated_.bind(this),
+    )
+  }
+
+  onContainersListUpdated_(containers: Container[]) {
+    this.containersList_ = containers
+  }
+
+  onAddContainerClick_() {
+    this.editingContainer_ = { id: '', name: '' }
+  }
+
+  onEditContainerClick_(e: DomRepeatEvent<Container>) {
+    // Copy the container to the editingContainer_ to avoid mutating the
+    // original container before saving.
+    this.editingContainer_ = { ...e.model.item }
+  }
+
+  onDeleteContainerClick_(e: DomRepeatEvent<Container>) {
+    this.deletingContainer_ = e.model.item
+  }
+
+  onCancelDialog_() {
+    this.editingContainer_ = null
+    this.deletingContainer_ = null
+  }
+
+  onSaveContainerFromDialog_() {
+    assert(this.editingContainer_)
+    if (!this.editingContainer_.id) {
+      this.browserProxy.handler.addContainer(this.editingContainer_)
+    } else {
+      this.browserProxy.handler.updateContainer(this.editingContainer_)
+    }
+    this.editingContainer_ = null
+  }
+
+  async onDeleteContainerFromDialog_() {
+    assert(this.deletingContainer_)
+
+    const dialog = this.shadowRoot!.querySelector(
+      '#deleteContainerDialog',
+    ) as CrDialogElement
+    assert(dialog)
+
+    const buttons = dialog.querySelectorAll('cr-button, cr-icon-button')
+    if (buttons) {
+      buttons.forEach((button) => button.setAttribute('disabled', 'true'))
+    }
+    dialog.noCancel = true
+
+    await this.browserProxy.handler.removeContainer(this.deletingContainer_.id)
+
+    if (buttons) {
+      buttons.forEach((button) => button.removeAttribute('disabled'))
+    }
+    dialog.noCancel = false
```

Comment:
Can we achieve this with a property that the UI binds to instead of manual DOM operations? We should only be doing this as a last resort.

Reply:
That is a question to you and @fallaciousreasoning 😅. I'm not aware of a simple way of doing this. I don't want to chromium_src-patch `CrDialog` in this PR if that's what you're suggesting.

I'm using the standard chromium `cr-dialog` here which doesn't offer any utility to disable it or its buttons, that's why I had to improvise here.